.github/workflows/test.yml

```diff
@@ -0,0 +1,50 @@
+name: Test Plugin
+
+on:
+  push:
+    branches-ignore:
+      - 'whitesource-remediate/**'
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
+        java:
+          - 17
+    if: github.repository == 'opensearch-project/opensearch-ai-flow-framework'
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ matrix.java }}
+          distribution: temurin
+
+      - name: Test
+        run: ./gradlew test
+
+      - name: YamlRestTest
+        run:
+          ./gradlew yamlRestTest
+
+      - name: integTest
+        run:
+          ./gradlew integTest
+
+      - name: Run Tests with Coverage
+        run:
+          ./gradlew test jacocoTestReport --info
+
+      - name: Upload Coverage Report
+        if: matrix.os == 'ubuntu-latest'
+        uses: codecov/codecov-action@v3
+        with:
+          file: ./build/reports/jacoco/test/jacocoTestReport.xml
```

Comment:
```suggestion
  test:
    if: github.repository == 'opensearch-project/opensearch-ai-flow-framework'
    strategy:
      matrix:
        test:
          - test jacocoTestReport
          - integTest
          - yamlRestTest
        os:
          - ubuntu-latest
          - macOS-latest
          - windows-latest
        java:
          - 17
    runs-on: ${{ matrix.os }}
    steps:
      - uses: actions/checkout@v4
      - name: Set up JDK ${{ matrix.java }}
        uses: actions/setup-java@v3
        with:
          java-version: ${{ matrix.java }}
          distribution: temurin
      - name: Test
        run: ./gradlew ${{ matrix.test }}
      - name: Upload Coverage Report
        if: matrix.os == 'ubuntu-latest' && matrix.test == 'test jacocoTestReport'
        uses: codecov/codecov-action@v3
        with:
          file: ./build/reports/jacoco/test/jacocoTestReport.xml
```

Reply:
You could have raised the PR :)

Reply:
It's probably not the right way to go, just pointing out how to achieve what you're trying to do :)